### ISSUE TITLE
[Unable to re-submit task]

### ISSUE THREAD

**Issue by govind-sah** [bactopia/bactopia#325]
Hi Bactopia community,
I am trying to analyze whole genome sequencing data for some of the _Pseudomonas aeruginosa_ isolates (72 to be specific). I haven't used bactopia before and tried to run the tool on my local computer following tutorial for `Multiple Samples (FOFN)` from this [bactopia github page](https://bactopia.github.io/tutorial/). 
My samples are in the fastq.gz format and were sequenced using paired end sequencing on MiSeq platform.
![Screen Shot 2022-05-28 at 5 10 38 PM](https://user-images.githubusercontent.com/49325989/170846681-f26fe8d4-60e1-4e42-b10a-6c3a33474727.png)

First I created FOFN file for multiple samples using the command
`bactopia prepare ./ > fastqs.txt`
the **fastqs.txt** file looks something like this

[fastqs.txt](https://github.com/bactopia/bactopia/files/8792378/fastqs.txt)

and then

I build **datasets** for **A Single Bacterial Species** using the following command
`bactopia datasets --species "Pseudomonas aeruginosa" --include_genus`

contents in the **datasets** are as shown below:
![Screen Shot 2022-05-28 at 5 18 01 PM](https://user-images.githubusercontent.com/49325989/170846848-374d93c0-85a4-4d34-950d-043e19de8ffb.png)

and finally, ran bactopia using the following command
```
bactopia --fastqs fastqs.txt \                                      
         --datasets datasets/ \
         --species "Pseudomonas aeruginosa" \
         --coverage 100 \
         --genome_size median \
         --outdir OUTPUT \
         --max_cpus 2
```

After running for around 20 minutes I got following message on terminal which I copied into a text file below.
[bactopia_error.txt](https://github.com/bactopia/bactopia/files/8792385/bactopia_error.txt)

Can someone please help me troubleshoot this or have any suggestion about how to analyze these genomes for SNPs or other kind of mutations.
Accept my apologies if my writing style is not professional or if it is too lengthy.
Thank you,
Govind

**Comment by rpetit3** [comment]
Hi @govind-sah ! 

First, please don't be concerned about your writing style or length. I had see no issues with it and actually found the length very useful. 

Look at the log, I'm wondering if you ran out of space. How much free space do you have left on your hard drive?

Cheers,
Robert

**Comment by govind-sah** [comment]
I have a macMini with 8 gb ram, 3 GHz 6-Core Intel Core i5 processor and only 256 gb hard drive. I was experiencing the storage issue earlier so I moved the raw fastq files (which is ~65 gb) to an external hard drive (fastqs.txt has the path to these files). I had more than 80 gb free space before running the bactopia command but as you pointed, I have only 2 gb space left on the internal hard drive (which I used as the working directory, unfortunately :( ). I'll try to rerun the command by changing the output directory to the same external drive (1TB). I hope it works.
And thank you so much for your prompt response despite posting my query on the weekend.

**Comment by rpetit3** [comment]
No worries! Hope the external drive does the trick for you. If not, you can try doing it in smaller batches. The `work` directory can sometimes take up  a lot of space when running in large batches.

Keep me posted! And please let me know if there are other issue you encounter. I no longer have access to a Mac system, but I'll try my best!

Cheers,
Robert

**Comment by govind-sah** [comment]
Hi @rpetit3,
I tried to rerun bactopia using just 3 samples (6 fastq.gz files total) this time and still getting error message (see below).
[error_message.txt](https://github.com/bactopia/bactopia/files/8794146/error_message.txt).
Can you please tell me what's going wrong here.
Alternatively, do you suggest using container for bactopia? I haven't used it before but I can give it a try if the problem is on my system. Please share the image for the latest version of bactopia if you have it.
Thank you,
Govind

**Comment by rpetit3** [comment]
If possible I would definitely recommend trying Docker.

Looks like its an issue with Conda:
```
bash -c 'mamba env create -f /Users/pridelab/opt/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/mac/sequence_type.yml --prefix /Users/pridelab/opt/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x --force'\
\
Standard output:\
conda-forge/osx-64                                          Using cache\
conda-forge/noarch                                          Using cache\
bioconda/osx-64                                             Using cache\
bioconda/noarch                                             Using cache\
r/osx-64                                                    Using cache\
r/noarch                                                    Using cache\
\
\
Looking for: ['ariba=2.14.6', 'bcftools=1.14', 'beautifulsoup4=4.10.0', 'biopython=1.79', 'blast=2.11.0', 'bowtie2=2.3.5.1', 'brotli=1.0.9', 'brotli-bin=1.0.9', 'bzip2=1.0.8', 'c-ares=1.18.1', 'ca-certificates=2021.10.8', 'cd-hit=4.8.1', 'certifi=2021.10.8', 'coreutils=9.0', 'curl=7.81.0', 'cycler=0.11.0', 'dendropy=4.5.2', 'entrez-direct=16.2', 'fonttools=4.29.1', 'freetype=2.10.4', 'gettext=0.19.8.1', 'gsl=2.7', 'hmmer=3.3.2', 'htslib=1.14', 'jpeg=9e', 'kiwisolver=1.3.2', 'krb5=1.19.2', 'lcms2=2.12', 'libblas=3.9.0', 'libbrotlicommon=1.0.9', 'libbrotlidec=1.0.9', 'libbrotlienc=1.0.9', 'libcblas=3.9.0', 'libcurl=7.81.0', 'libcxx=12.0.1', 'libdeflate=1.9', 'libedit=3.1.20191231', 'libev=4.33', 'libffi=3.4.2', 'libgfortran=5.0.0', 'libgfortran5=9.3.0', 'libiconv=1.16', 'libidn2=2.3.2', 'liblapack=3.9.0', 'libnghttp2=1.46.0', 'libopenblas=0.3.18', 'libpng=1.6.37', 'libssh2=1.10.0', 'libtiff=4.2.0', 'libunistring=0.9.10', 'libwebp-base=1.2.2', 'libzlib=1.2.11', 'llvm-openmp=13.0.1', 'lz4-c=1.9.3', 'matplotlib-base=3.5.1', 'mummer=3.23', 'munkres=1.1.4', 'ncbi-amrfinderplus=3.10.23', 'ncurses=6.2', 'numpy=1.21.5', 'olefile=0.46', 'openjpeg=2.4.0', 'openssl=3.0.0', 'packaging=21.3', 'pcre=8.45', 'perl=5.26.2', 'perl-archive-tar=2.32', 'perl-carp=1.38', 'perl-common-sense=3.74', 'perl-compress-raw-bzip2=2.087', 'perl-compress-raw-zlib=2.087', 'perl-exporter=5.72', 'perl-exporter-tiny=1.002001', 'perl-extutils-makemaker=7.36', 'perl-io-compress=2.087', 'perl-io-zlib=1.10', 'perl-json=4.02', 'perl-json-xs=2.34', 'perl-list-moreutils=0.428', 'perl-list-moreutils-xs=0.428', 'perl-pathtools=3.75', 'perl-scalar-list-utils=1.52', 'perl-types-serialiser=1.0', 'perl-xsloader=0.24', 'pigz=2.6', 'pillow=8.2.0', 'pip=22.0.3', 'pyfastaq=3.17.0', 'pymummer=0.10.3', 'pyparsing=3.0.7', 'pysam=0.18.0', 'python=3.7.12', 'python-dateutil=2.8.2', 'python_abi=3.7', 'readline=8.1', 'samtools=1.14', 'sed=4.7', 'setuptools=60.9.3', 'six=1.16.0', 'soupsieve=2.3.1', 'spades=3.15.2', 'sqlite=3.37.0', 'tbb=2021.5.0', 'tk=8.6.12', 'unicodedata2=14.0.0', 'wget=1.20.3', 'wheel=0.37.1', 'xz=5.2.5', 'zlib=1.2.11', 'zstd=1.5.2']\
\
\
Encountered problems while solving:\
  - package bowtie2-2.3.5.1-py27h2dec4b4_0 requires python >=2.7,<2.8.0a0, but none of the providers can be installed\
```

I might be able to update the dependencies if you don't think you'll be able to use Docker.

**Comment by govind-sah** [comment]
I will try to use docker for this however, since I am not very familiar with it, it am afraid if it will work for me.
If possible, please update the dependencies so that I can use it on my system.
Thank you so much for your assistance and guidance.

**Comment by govind-sah** [comment]
Hi @rpetit3,
I tried to run bactopia in docker this time, but was unsuccessful.
I just used one sample (S14) for the analysis  and got the following errors.
[Docker_bactopia_error.txt](https://github.com/bactopia/bactopia/files/8839004/Docker_bactopia_error.txt)

I'm not sure why it is not working no matter what I try.
Is there some issue on my computer?
Please help.
Thank you,
Govind

**Comment by rpetit3** [comment]
You are good.  You will want to run bactopia from the conda environment then add `-profile docker` to tell Nextflow to use Docker.

```
conda activate bactopia
bactopia --fastqs fastqs.txt \                                      
         --datasets datasets/ \
         --species "Pseudomonas aeruginosa" \
         --coverage 100 \
         --genome_size median \
         --outdir OUTPUT \
         --max_cpus 2
         -profile docker
```

Let me know if this works for you

**Comment by govind-sah** [comment]
Apparently, this seem to work initially as it ran for some time but eventually gave me an error :(.
My terminal looked something like this (see the file below).
[bactopia_errors.docx](https://github.com/bactopia/bactopia/files/8841092/bactopia_errors.docx)

thanks,

**Comment by rpetit3** [comment]
I think `S14` is requiring more than 8GB of memory to process. 

How big are the fastqs for `S14`?

**Comment by govind-sah** [comment]
Sample S14 or any other is not huge. they are in the range of 400-500 mb each file. 
S14_R1.fastq.gz is 483.3 mb and S14_R2.fastq.gz is 525.2 mb in size.
I tried to run another sample (S15, which are 400(R1) and 433(R2) mb in size) using the following command:
`bactopia --R1 S15_R1.fastq.gz --R2 S15_R2.fastq.gz --sample S15 --datasets datasets/ --species "Pseudomonas aeruginosa"  --coverage 100 --genome_size median --outdir local-single-sample --max_cpus 2  -profile docker`
but again got error. See the file below.
[errors.docx](https://github.com/bactopia/bactopia/files/8850670/errors.docx)


Thank you,
Govind

**Comment by rpetit3** [comment]
I wonder if there is a setting in Docker to increase the amount of ram"

```
  [shovill] Machine has 3 CPU cores and 3.84 GB RAM
  [shovill] Set --ram to 7 but machine only has 3.84 GB
```

Maybe this will help: https://docs.docker.com/desktop/mac/

**Comment by govind-sah** [comment]
Aaahhhh😓....finally it worked😊!
Although some modules did not run, I suppose. This time I used Macbook pro (3.1 GHz Dual-Core Intel Core i7; 16 GB 1867 MHz DDR3) and it took around 7 hours to finish the run. I wonder 🥺 how much time will it take to run all my samples!

This time I also changed the preference of Docker desktop (see below) as you suggested.
<img width="1276" alt="Screen Shot 2022-06-07 at 9 56 25 PM" src="https://user-images.githubusercontent.com/49325989/172534964-1abaa62d-1e8c-4802-968f-ee448802a5d4.png">

I am being curious 🤔 for why do I need so much memory even if I use docker as this same run failed earlier today on my previous system (MacMini with 8GB ram) but completed on this system with 16GB ram.

 Also, is it normal to see some module failing?

Here is what I see on my terminal.
[Bactopia_run_output.docx](https://github.com/bactopia/bactopia/files/8858449/Bactopia_run_output.docx)


Thank you so much for answering all my queries and helping me get through.
I appreciate your help🙏.
Govind

**Comment by rpetit3** [comment]
The failed steps are ok as long as the retry passes. I think 6 hours is kind of long, how was your internet connection? 

I ask because the first time you run bactopia its going to spend a lot of time pulling the docker images (~7gb) I usually expect samples to finish processing in 30-60 minutes.

**Comment by govind-sah** [comment]
I have a fairly good internet connection ~200mbps.
I will try to run other samples too and let you know the results.
Also, I did not understand why it needs so much of ram on my system if running through the docker. Do you know the reason?
Thank you,

**Comment by rpetit3** [comment]
Hi @govind-sah,

Unfortunately I'm not too familiar with Docker on Mac. But in other new, I released v2.1.0 (https://github.com/bactopia/bactopia/releases/tag/v2.1.0) which might fix your initial issue using Conda. 

If you have the time, I'm very curious to know if v2.1.0 allows you to use Conda. 

Cheers,
Robert

**Comment by govind-sah** [comment]
I used the latest version of bactopia to analyze of my samples, S14.
Apparently, it improved compared to the earlier version since I can see better results, however not all modules ran successfully.
Please have a look at my terminal output (view the file below).
[bactopia_terminal.docx](https://github.com/bactopia/bactopia/files/8866057/bactopia_terminal.docx)

thank you,
Govind

**Comment by rpetit3** [comment]
Bummer! I've seen something similar, going to try an replicate on my end. Will update shortly

**Comment by govind-sah** [comment]
@rpetit3 Sorry but I could not figure it out on my own.
It still gives me an error when running multiple samples together.
Did you update it recently?
Thanks,
Govind

**Comment by rpetit3** [comment]
Hi @govind-sah 

I've released [v2.1.1](https://github.com/bactopia/bactopia/releases/tag/v2.1.1), I'm curious if it still gives you the same issue

Cheers,
Robert

**Comment by rpetit3** [comment]
Hi @govind-sah 

I'm going to go ahead and close this issue, please feel free to reopen

Cheers,
Robert
